src/Lecture4.hs

```diff
@@ -226,18 +327,49 @@ implement the next task.
 -}
 
 combineRows :: NonEmpty Row -> Stats
-combineRows = error "TODO"
+combineRows rows = case rows of
+    x :| [] -> rowToStats x
```

Comment:
Strictly speaking, this pattern is redundant because `foldl'` already handles the case with empty list 🙂